spring-amqp/src/main/java/org/springframework/amqp/support/SimpleAmqpHeaderMapper.java

```diff
@@ -71,6 +71,8 @@ public void fromHeaders(MessageHeaders headers, MessageProperties amqpMessagePro
 		javaUtils
 			.acceptIfNotNull(getHeaderIfAvailable(headers, AmqpHeaders.DELAY, Integer.class),
 					amqpMessageProperties::setDelay)
```

Comment:
I think this one has to call `setDelayLong` as well.
We indeed cannot guarantee that other side didn't set this header as an integer, but internally we still must work with non-deprecated API.